lib/temporal/thread_pool.rb

```diff
@@ -20,6 +21,11 @@ def initialize(size)
       end
     end
 
+    def report_metrics
+      Temporal.metrics.gauge(Temporal::MetricKeys::THREAD_POOL_AVAILABLE_THREADS, @available_threads, @metrics_tags)
+      Temporal.metrics.gauge(Temporal::MetricKeys::THREAD_POOL_QUEUE_SIZE, @queue.size, @metrics_tags)
```

Comment:
The queue size metric isn't very useful and will probably always be zero or 1. This is due to the fact that our pollers are blocking on `wait_for_available_threads` before polling for the next task. So the queue is just a transmission mechanism to communicate with pool threads and not ac actual buffer.

What is useful is an already existing metric called `time_since_last_poll` which will tell you if your thread pool is busy and the worker is unable to poll for tasks for extended periods of time.

Reply:
That's fair. We had inspected all these metrics when diagnosing a performance issue, but queue size turned out to not be as useful. I'll remove it.